packages/svelte-docgen/src/doc/type.ts

```diff
@@ -212,8 +213,42 @@ export interface TypeParam {
 	default?: TypeOrRef;
 }
 
-export interface Union extends WithAlias {
-	kind: "union";
+export interface Index {
+	kind: "index";
+	type: TypeOrRef;
+}
+
+export interface IndexedAccess {
+	kind: "indexed-access";
+	objectType: TypeOrRef;
+	indexType: TypeOrRef;
+	constraint?: TypeOrRef;
+	simplifiedForReading?: TypeOrRef;
+	simplifiedForWriting?: TypeOrRef;
```

Comment:
I couldn't find in the tests snapshots how they look like. How they are different?

Reply:
I also don’t understand what these are supposed to represent… 😓

Reply:
I'm about to finish reviewing. This one is unresolved.

Are we going to keep them? Because if I have to document those fields, I wouldn't have a clue how to explain them. 😅

Reply:
Okay, I’ve commented these out for now and left a TODO comment.